### ISSUE TITLE
Custom health indicator fail does not affect main status

### ISSUE THREAD

**Issue by TrejGun** [nestjs/terminus#1284]
## I'm submitting a...
<pre><code>
[ ] Regression
[x ] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior
Here is my health.controller.ts

```ts
   private readonly myService: myServiceHealthIndicator,

  @Get()
  @HealthCheck()
  readiness(): Promise<HealthCheckResult> {
    const rmqUrl = this.configService.get<string>("RMQ_URL", "amqp://127.0.0.1:5672/");

    return this.health.check([
      async (): Promise<HealthIndicatorResult> =>
        this.db.pingCheck("database", {
          timeout: 300,
        }),
      async (): Promise<HealthIndicatorResult> => this.myService.isHealthy("myService"),
    ]);
  }
```

Here is my myServiceHealthIndicator

```ts
  async isHealthy(key: string): Promise<HealthIndicatorResult> {
    return this.httpService
      .request({
        url: "http://localhost:3004/health",
      })
      .pipe(map((response: {data: HealthCheckResult}) => response.data))
      .toPromise()
      .then(...check for status "UP"...)
      .catch(() => {
        return this.getStatus(key, false);
      });
  }
```

health.controller does check for two things - database and microservice
myServiceHealthIndicator check HealthCheckResult from microservice

when HealthCheckResult from myService is  "down"
```json
{
    ...
    "myService": {
      "status": "down"
    }
}
```

the overall status of health.controller is "OK"
```json
{
  "status": "ok",
  "info": {
    "database": {
      "status": "up"
    },
    "myService": {
      "status": "down"
    }
  },
  "error": {},
  "details": {
    "database": {
      "status": "up"
    },
    "myService": {
      "status": "down"
    }
  }
}
```

## Expected behavior
If any health indicator failed the overall status must be "error"

## Minimal reproduction of the problem with instructions
see above

## Environment

<pre><code>
Nest version: nest core 7.6.17, terminus 7.2.0

 
For Tooling issues:
- Node version: v14.17.0  
- Platform:  mac 

Others:

Others:
</code></pre>

**Comment by BrunnerLivio** [comment]
Great catch, let me look into that!

**Comment by BrunnerLivio** [comment]
Terminus does explicitly check for errors of class `HealthCheckError`. See more [the example of the docs](https://docs.nestjs.com/recipes/terminus#custom-health-indicator). The idea behind it is that Terminus does not accidentally display an error message which might expose confidential information -- therefore we don't "catch everything".

So you can fix this by using:

```
 async isHealthy(key: string): Promise<HealthIndicatorResult> {
    return this.httpService
      .request({
        url: "http://localhost:3004/health",
      })
      .pipe(map((response: {data: HealthCheckResult}) => response.data))
      .toPromise()
      .then(...check for status "UP"...)
      .catch(() => {
        throw throw new HealthCheckError('Healthcheck failed', this.getStatus(key, false));
      });
  }
```

You can also use the [`HttpHealthIndicator.responseCheck`](https://github.com/nestjs/terminus/blob/master/lib/health-indicator/http/http.health.ts#L85) where you can pass a custom validator function to check whether the response is healthy.

```typescript
return this.health.check([
  () => this.http.responseCheck('nestjs-docs', 'https://docs.nestjs.com', (res) => res.statusCode !== 404),
]);

**Comment by TrejGun** [comment]
the problem with `responseCheck` was - it passes error code from ms (http://localhost:3004/health) to main service so the resulting JSON was fine but the HTTP status was wrong

**Comment by BrunnerLivio** [comment]
Really sorry for the delay!

@TrejGun not quite sure what you're referring to!

**Comment by TrejGun** [comment]
the microservice I tried to check failed with 503 and this status was passed all the way back to the app and the health check of the app returned the same status instead of 200. 

anyways to ping ms by HTTP is not an option. we tried on production and realized that we have an autoscale group and all machines in the group try to ping ms so we have about 10 health check requests per second. so we gave up on this approach
